lib/spack/docs/packages_yaml.rst

```diff
@@ -383,20 +383,52 @@ like this:
 
 which means every spec will be required to use ``clang`` as a compiler.
 
-Note that in this case ``all`` represents a *default set of requirements* -
+Any constraint that appears in default requirements under ``all``, and that
+would clearly lead to unsolvable cases, is disregarded. For instance a configuration
+like the following:
```

Comment:
```suggestion
Requirements on variants for all packages are possible too, but note that they
are only enforced for those packages that define these variants, otherwise they
are disregarded. For example:
```

I think it's important to stress "variants" here, cause there are other constraints that can "clearly" lead to unsolvable cases like `@1:`.